subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/ResolveConfigurationResolutionBuildOperationResult.java

```diff
@@ -18,37 +18,27 @@
 
 import com.google.common.collect.Maps;
 import org.gradle.api.Action;
+import org.gradle.api.artifacts.result.ResolutionResult;
 import org.gradle.api.artifacts.result.ResolvedComponentResult;
 import org.gradle.api.internal.artifacts.result.ResolvedComponentResultInternal;
-import org.gradle.internal.Actions;
 import org.gradle.internal.operations.trace.CustomOperationTraceSerialization;
 
 import java.util.Collections;
 import java.util.HashMap;
 import java.util.Map;
 
 class ResolveConfigurationResolutionBuildOperationResult implements ResolveConfigurationDependenciesBuildOperationType.Result, CustomOperationTraceSerialization {
+    private final ResolvedComponentResult root;
+    private final ResolutionResult resolutionResult;
 
-    private static final Action<? super Throwable> FAIL_SAFE = Actions.doNothing();
-
-    private final ResolvableDependenciesInternal incoming;
-    private final boolean failSafe;
-
-    ResolveConfigurationResolutionBuildOperationResult(ResolvableDependenciesInternal incoming, boolean failSafe) {
-        this.incoming = incoming;
-        this.failSafe = failSafe;
+    ResolveConfigurationResolutionBuildOperationResult(ResolutionResult resolutionResult) {
+        this.root = resolutionResult.getRoot();
```

Comment:
We shouldn't do this until it's queried, otherwise we're deserialising the dependency graph from disk when the result event is fired, regardless of whether the graph required or not. It won't be required, for example, if the build scan plugin isn't applied.

Also, we should raise an issue somewhere to avoid the serialise - deserialise step when the build scan plugin is applied, but we won't have made this any worse than it was.

Reply:
Brilliant, deferring seems to keep all the test happy. A CI run has been started.